src/Processors/QueryPlan/Streaming/ReplayStreamStep.cpp

```diff
@@ -22,14 +22,16 @@ static ITransformingStep::Traits getTraits()
         }};
 }
 
-ReplayStreamStep::ReplayStreamStep(const DataStream & input_stream_, Float32 replay_speed_)
-    : ITransformingStep(input_stream_, input_stream_.header, getTraits()), replay_speed(replay_speed_)
+ReplayStreamStep::ReplayStreamStep(const DataStream & input_stream_, Float32 replay_speed_, const std::vector<Int64> & shards_last_sn_)
+    : ITransformingStep(input_stream_, input_stream_.header, getTraits()), replay_speed(replay_speed_), shards_last_sn(shards_last_sn_)
```

Comment:
You can pass `std::vector<Int64> shards_last_sns_`, and initialize as `shards_last_sns(std::move(shards_last_sns_))`

Reply:
ok, This will save one copy operation？